File: apps/backend/internal/interface/http/handler/metadata/dto/metadata.go
package metadatadto

type CompanySize string

const (
	CompanySizeSmall  CompanySize = "small"
	CompanySizeMedium CompanySize = "medium"
	CompanySizeLarge  CompanySize = "large"
)

type CountryMetadata struct {
	Code            string   `json:"code"`
	Label           string   `json:"label"`
	Currency        string   `json:"currency"`
	DefaultTimezone string   `json:"defaultTimezone"`
	Timezones       []string `json:"timezones"`
}

type IndustryMetadata struct {
	Name     string   `json:"name"`
	Keywords []string `json:"keywords,omitempty"`
}

type TenantMetadataPayload struct {
	Industries   []IndustryMetadata `json:"industries"`
	CompanySizes []CompanySize      `json:"companySizes"`
	Countries    []CountryMetadata  `json:"countries"`
}

File: apps/frontend/types/metadata.ts
export interface Industry {
  name: string;
  keywords?: string[];
}

export interface Country {
  code: string;
  label: string;
  currency: string;
  defaultTimezone: string;
  timezones: string[];
}

export interface TenantMetadataConfig {
  industries: Industry[];
  companySizes: string[];
  countries: Country[];
}
